gpii/node_modules/matchMakerFramework/src/MatchMakerFramework.js

```diff
@@ -103,6 +110,150 @@
+	 * Returns NULL if no matchmaker preference was found.
+     */
+	gpii.matchMakerFramework.GetPreferredMatchMaker = function (matchMakerPayload) {
+		// Rejection tests
+		if (matchMakerPayload.preferences === undefined || matchMakerPayload.preferences.contexts === undefined || matchMakerPayload.activeContexts === undefined) {
+			return null;
+		}
+		// Search for mm selection preferences
+		var contexts = matchMakerPayload.preferences.contexts;
+		var activeContexts = matchMakerPayload.activeContexts;
+		var foundPriority = -1;
+		var foundMatchMaker;
+		fluid.each(activeContexts, function (activeContext) {
+			if (contexts[activeContext] && contexts[activeContext].preferences) {
+				var prefs = contexts[activeContext].preferences;
+				var curPriority = 0;
+				if (contexts[activeContext].priority) {
+					curPriority = contexts[activeContext].priority;
+				}
+				if ((prefs["http://registry.gpii.net/common/matchMakerType"]) && (curPriority > foundPriority)) {
+					foundPriority = curPriority;
+					foundMatchMaker = prefs["http://registry.gpii.net/common/matchMakerType"];
+				}
+			}
+		});
+		return foundMatchMaker;
+    };
+
+    /*
+	 * Returns the Matchmaker that best matches to the playload based on its capabilities.
+	 * See MatchMakerCapabilities.json.
+     */
+	gpii.matchMakerFramework.GetCapableMatchMaker = function (matchMakerPayload) {
+		// Contexts
+		var contexts = matchMakerPayload.preferences.contexts;
+		// Find excluded matchmakers
+		var excludedMatchMakers = [];
+		var activeContexts = matchMakerPayload.activeContexts;
+		fluid.each(activeContexts, function (activeContext) {
+			if (contexts[activeContext] && contexts[activeContext].preferences) {
+				var prefs = contexts[activeContext].preferences;
+				if (prefs["http://registry.gpii.net/common/excludedMatchMakers"]) {
+					excludedMatchMakers = prefs["http://registry.gpii.net/common/excludedMatchMakers"];
+				}
+			}
+		});
+		// Try again block
+		if ( matchMakerPayload.tryAgain && matchMakerPayload.tryAgain.active) {
+			if (matchMakerPayload.tryAgain[excludedMatchMakers]) {
+				excludedMatchMakers.concat(matchMakerPayload.tryAgain.excludedMatchMakers);
+			}
+		}
+		// Calculate metric values
+		var payloadHasConditions = false;
+		var payloadPreferenceCount = 0;
+		fluid.each(contexts, function (context) {
+			if (context.conditions && (context.conditions.length > 0)) {
+				payloadHasConditions = true;
+			}
+			if (context.preferences) {
+				fluid.each(context.preferences, function (pref) {
+					if (typeof pref == "object") {
+						payloadPreferenceCount += Object.keys(pref).length;
+					} else {
+						++payloadPreferenceCount;
+					}
+				});
+			}
+		});
+		// Compute capabilities
+		var foundMatchmaker = false;
+		var confidence = {};
+		fluid.each(gpii.matchMakerFramework.matchmakerCapabilities, function (capability) {
+			if (excludedMatchMakers.indexOf(capability.id) < 0) {
```

Comment:
could you call the `id` in capabilities `matchMakerId` or something like that... Down here the straight `id` reference becomes quite confusing, and one think it refers to the id of the capability